sdc/tests/test_dataframe.py

```diff
@@ -813,16 +813,26 @@ def test_impl(df):
     def test_df_isna(self):
         def test_impl(df):
             return df.isna()
+
         sdc_func = sdc.jit(test_impl)
-        for idx in [[3, 4, 2, 6, 1], ['a', 'b', 'c', 'd', 'e'], None]:
-            df = pd.DataFrame({"FLOAT": [3.2, np.nan, 7.0, 3.3, np.nan],
-                                "INT": [3, 4, 1, 0, 222],
-                                "BOOL": [True, True, False, False, True],
-                                "STRING": ['a', 'dd', 'c', '12', None]
-                                }, index=idx)
+        indexes = [[3, 4, 2, 6, 1], ['a', 'b', 'c', 'd', 'e'], None]
+
+        for idx in indexes:
+            df = pd.DataFrame({"float": [3.2, np.nan, 7.0, 3.3, np.nan],
+                               "int": [3, 4, 1, 0, 222],
+                               "string": ['a', 'dd', 'c', '12', None]}, index=idx)
             with self.subTest(index=idx):
                 pd.testing.assert_frame_equal(sdc_func(df), test_impl(df))
 
+    @unittest.skip('DF with bool data Segmentation fault')
+    def test_df_bool(self):
```

Comment:
Unboxing bool column of a DF and calling isna on it works fine, so the bools are not the reason here. It seems the problem is linked with column name "bool", so I would advise just renaming the columns to ordinary names 'A', 'B', 'C', etc and add the bool column into the test test_df_isna. This way it would be complete.
As for the failing test let's rewrite and rename it so that it's clear we have an issue with such column names (you can also dig a bit deeper into the problem and try to locate the root cause and mention it in test skip reason).